app/app/settings.py

```diff
@@ -74,12 +74,21 @@
     'django.middleware.common.CommonMiddleware',
     'django.middleware.csrf.CsrfViewMiddleware',
     'django.contrib.auth.middleware.AuthenticationMiddleware',
+    'django.contrib.auth.middleware.RemoteUserMiddleware',
     'django.contrib.messages.middleware.MessageMiddleware',
     'django.middleware.clickjacking.XFrameOptionsMiddleware',
     'social_django.middleware.SocialAuthExceptionMiddleware',
     'applicationinsights.django.ApplicationInsightsMiddleware',
+    'custom_auth.CustomSSOMiddleware', 
 ]
 
+# added for custom SSO 
+CREATE_USER = env.bool("CREATE_USER", default=True)
+IS_SUPER = env.bool("IS_SUPER", default=False)
+HEADER_AUTH_PROXY_USER='HTTP_X_AUTH_PROXY_USER'
```

Comment:
Header values should be configurable via environment variables to make it easier to customize deployments.

Reply:
Configure header values for ```HEADER_AUTH_PROXY_USER```? Not sure I follow.

Reply:
The `HEADER_AUTH_PROXY_USER` variable holds the name of the header that the system uses to check for the name of the user that's being logged in, correct? If so, we should be able to configure this header name via an environment variable for each doccano deployment since likely each auth proxy will use a different header value.

```py
HEADER_AUTH_PROXY_USER = env('AUTH_PROXY_USER_HEADER')
```

Reply:
tl;dr, no value in `settings.py` should be hard-coded, everything should be configurable via the environment so that the stack is fully customizable (see [12-factor app methodology](https://12factor.net/config)).